src/Hooks/RoleValidator.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Pantheon\Terminus\Hooks;
+
+use Consolidation\AnnotatedCommand\CommandData;
+use Pantheon\Terminus\Exceptions\TerminusException;
+
+class RoleValidator
+{
+    const ORG_ROLES = 'admin|developer|team_member|unprivileged';
+    const PARAM_NAME = 'role';
+    const SITE_ROLES = 'developer|team_member';
+
+    /**
+     * @hook validate *
+     *
+     * @param CommandData $command_data
+     * @throws TerminusException If the input role is invalid
+     */
+    public function validateRole(CommandData $command_data)
+    {
+        $input = $command_data->input();
+        if (!$input->hasArgument(self::PARAM_NAME)) {
```

Comment:
Then we narrow it down to only commands with this parameter. If the parameter was supposed to be provided and it wasn't it'll err before it reaches this hook.